### PR TITLE
SAK-48871 tasks. Fix completed checkbox issues

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-48871

### CODE REVIEW

File: webcomponents/tool/src/main/frontend/js/sakai-date-fns.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  if (options.lang) {
+    if (options.lang.includes("-")) {
+      const parts = options.lang.split("-");
+      // Try the langcountry key, but fallback to just the language if that doesn't exist
+      options.locale = locales[parts.join("")] || locales[parts[0]];
+    } else {
+      options.locale = locales[options.lang];
+    }
+  } else {
+    options.locale = locales[enUS];
```
Comment (ottenhoff) [inline webcomponents/tool/src/main/frontend/js/sakai-date-fns.js:20]: Can you move to the default to the top instead? 

options.locale = locales[enUS];
if (options.lang) {

What happens if both of these are null?

locales[parts.join("")] || locales[parts[0]];

**Comment by ottenhoff** [comment]
are you replacing momentjs? i don't love having another date-manipulation library just for this one function. moment seems to do what you're looking to do:

`moment().startOf('hour').fromNow();   `

**Comment by adrianfish** [comment]
Moment is deprecated, and heavy. I'm only replacing it for tasks.

**Comment by ottenhoff** [comment]
> Moment is deprecated, and heavy. I'm only replacing it for tasks. I was using moment here, just to humanize the durations.

the manual locale loading is 🤷

**Comment by adrianfish** [comment]
Tree shaking ... Why load all moment's locales? It bloats your bundles

**Comment by ottenhoff** [comment]
> Tree shaking ... Why load all moment's locales? It bloats your bundles. Good point though - I hate it too

because we are a big multi-locale app that supports you switching locales

**Comment by ottenhoff** [comment]
i guess it would be annoying if we were adding lots of locales, but we aren't so, maybe not a big deal. it's not fun from maintainability perspective, but hey ....

**Comment by adrianfish** [comment]
Moment is dead though. So we need to move to something at some point. We can do a lot of what moment does in vanilla JS, nowadays, but that humanize stuff is really nice and not in vanilla.

**Comment by adrianfish** [comment]
I've added all our locales to the module, and tested each of them in the tasks widget. Just to be clear, this PR will only move tasks to using date-fns. Moment will still be used elsewhere, same as now.

File: webcomponents/tool/src/main/frontend/js/sakai-date-fns.js (hunk 1 above)
Comment (ottenhoff) [reply webcomponents/tool/src/main/frontend/js/sakai-date-fns.js:24]: can you move the default up and out of the else statement?

**Comment by ottenhoff** [comment]
okay, right on ... let's move forward

**Comment by adrianfish** [comment]
@ottenhoff Go on, mash that approve button.

**Review by ottenhoff — APPROVED** [review]

**Pull request merged**